src/notify.js

```diff
@@ -71,6 +83,11 @@ Notify.isSupported = function(perm) {
         throw new Error('You must only call this before calling Notification.requestPermission(), otherwise this feature detect would trigger an actual notification!');
     }
 
+    if (Notify.serviceWorkerSupported()) {
```

Comment:
That could work, but I still may favor explicit feature detects for each API, as opposed to something transparent and not immediately obvious to what is and isn't "supported".